packages/api/src/store/rabbit-queue.ts

```
@@ -0,0 +1,104 @@
+import { Channel, ConsumeMessage, Options } from "amqplib";
+
+const QUEUE_NAME = "webhook_default_queue";
+const EXCHANGE_NAME = "webhook_default_exchange";
+
+export default class MessageQueue {
+  private channel: ChannelWrapper;
+  private connection: AmqpConnectionManager;
+
+  constructor() {}
+
+  public async connect(url: string): Promise<void> {
+    // Create a new connection manager
+    this.connection = await amqp.connect([url]);
+    this.connection.on("connect", () => console.log("AMQP Connected!"));
+    this.connection.on("disconnect", (err: Error) =>
+      console.log("AMQP Disconnected.", err)
+    );
+    this.connection.on("error", (error: Error) => {
+      console.error(`Connection error: ${error}`);
+    });
+
+    this.channel = await this.connection.createChannel({
+      json: true,
+      setup: async function (channel: Channel) {
+        await Promise.all([
+          channel.assertQueue(QUEUE_NAME),
+          // channel.assertQueue("delayedQueue", {
+          //   messageTtl: 5000,
+          //   deadLetterExchange: EXCHANGE_NAME
+          // }),
+          channel.assertExchange(EXCHANGE_NAME, "topic", { durable: true }),
+          channel.prefetch(1),
+          channel.bindQueue(QUEUE_NAME, EXCHANGE_NAME, "#"),
+          // channel.consume(QUEUE_NAME, handleMessage)
+        ]);
+        // return channel.assertQueue('rxQueueName', {durable: true});
+      },
+    });
+  }
+
+  public async close(): Promise<void> {
+    await this.connection.removeAllListeners("connect");
+    await this.connection.removeAllListeners("disconnect");
+    await this.connection.removeAllListeners("error");
+    await this.connection.close();
+  }
+
+  public ack(data: any): void {
+    this.channel.ack(data);
+  }
+
+  public async consume(func: (msg: ConsumeMessage) => void): Promise<void> {
+    if (!func) {
+      console.log("func is undefined");
+      func = this.handleMessage.bind(this);
+    }
```

From what I understood of the logic here, I am thinking that this default may be unnecessary as it may hide some coding error. WDYT of throwing an error here instead?